packages/react/src/utilities/ThemeProvider/makeStyles.ts

```diff
@@ -61,15 +62,38 @@ export function makeStyles<TStyleSet extends { [key in keyof TStyleSet]: IStyle
   // eslint-disable-next-line deprecation/deprecation
   return (options: UseStylesOptions = {}) => {
     let { theme } = options;
-    const win = useWindow();
+    let winId: string | undefined;
+    // Retain a dictionary of window ids we're tracking
+    const allWindows = new Set<string>();
+    const win = useWindow() as (Window & { __id__: string }) | undefined;
+    if (win) {
+      win.__id__ = win.__id__ || getId();
+      winId = win.__id__;
+      if (!allWindows.has(winId)) {
+        allWindows.add(winId);
+        win.addEventListener('unload', () => cleanupMapEntries(winId));
+      }
+      // cleanupMapEntries will
+      // 1. remove all the graph branches for the window,
+      // 2. remove the event listener,
+      // 3. delete the allWindows entry.
+      const cleanupMapEntries = (id: string | undefined) => {
```

Comment:
This can move up above the return; otherwise I think this will get redefined on every hook call on every render.